pxd_fastpath.c

```diff
@@ -912,6 +926,41 @@ bool pxd_sync_work_pending(struct pxd_device *pxd_dev)
 	return busy;
 }
 
+// external request to initiate failover on fastpath device
+int pxd_request_failover(struct pxd_device *pxd_dev)
+{
+       struct pxd_fastpath_extension *fp = &pxd_dev->fp;
+
+       // incompat device or already in native path
+       if (pxd_dev->using_blkque || !fp->fastpath) {
+               printk("device %llu failover request failed (blkque %d, fastpath %d)\n",
+                       pxd_dev->dev_id, pxd_dev->using_blkque, fp->fastpath);
+               return -EINVAL;
+       }
+
+       // IO path blocked, a future path refresh will take it to native path
+       // enqueue a failover request to userspace on this device.
+       printk("device %llu initiated failover\n", pxd_dev->dev_id);
+       return pxd_initiate_failover(pxd_dev);
+}
+
+// external request to initiate fallback on fastpath device
+int pxd_request_fallback(struct pxd_device *pxd_dev)
```

Comment:
pxd_request_failover/pxd_request_fallback can be collapsed into one.

Reply:
done, introduced pxd_request_ioswitch()